src/abi_1_0.c

```diff
@@ -122,6 +122,81 @@ struct fi_info_1_1 {
 	struct fi_fabric_attr_1_0	*fabric_attr;
 };
 
+struct fi_tx_attr_1_2 {
+        uint64_t                caps;
+        uint64_t                mode;
+        uint64_t                op_flags;
+        uint64_t                msg_order;
+        uint64_t                comp_order;
+        size_t                  inject_size;
+        size_t                  size;
+        size_t                  iov_limit;
+        size_t                  rma_iov_limit;
```

Comment:
> 
> @shefty 
> This is not a 1.2 field
> @jswaro
> jswaro 
> 
> I'm slightly confused. Perhaps you can walk me through this, but I thought 1.2 referred to the ABI version, not the API version.

Reply:
1.2 is the ABI version.  The structures here should refer to the current data structures (before any changes have been applied).  These are the structures that existing compiled applications will be using.  The addition of the tclass field is breaking the ABI, requiring the bump to 1.3.  The code in this file is responsible for converting from the 1.2 structures into the 1.3 structures.  This means that providers only need to deal with the 1.3 structures, with the core handling the conversion.

Reply:
I think I understood that part, but what was missing from the piece that I wrote? Just trying to identify the delta.

Reply:
The difference is that you included the tclass field in the 1_2 structure definitions.  Basically, you should copy struct fi_tx_attr to this file and rename as fi_tx_attr_1_2.  *Then* you add the new fields to struct fi_tx_attr (in fabric.h).

Reply:
I see now. Thanks!